src/MultiCollateralVault.sol

```diff
@@ -0,0 +1,346 @@
+    /// mapping(user => mapping(token => deposit))
+    mapping(address => mapping(address => uint256)) public Deposits;
+
+    /// @dev mapping for storage of Chainlink Oracle interfaces
+    /// @dev Token address pairings used as keys (ex mapping(BTC => mapping(ETH => AggregatorV3Interface)))
+    mapping(address => AggregatorV3Interface) public Oracles;
+
+    /// @dev mapping for storing the maximum LTV of a collateral token
+    /// @dev is a factor with 27 decimals
+    mapping(address => uint256) public Ratios;
+
+    address[] public Collaterals;
+
+    ///@param WETH_ address for the WETH contract
+    constructor (address WETH_) {
+        WETH = WETH_;
+    }
+
+/***
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+    Admin Functions
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+ */
+    function addCollateral(address token, AggregatorV3Interface oracle) external auth {
+        _addCollateral(token, oracle);
+    }
+
+    function setRatio(address token, uint256 rad) external auth {
+        _setRatio(token, rad);
+    }
+
+    function liquidate(address user) external auth {
+        _liquidate(user);
+    }
+
+    /// @dev checks to see if a token is currently accepted as collteral and if it has been added but not yet live cause there is no ratio set yet
+    function _addCollateral(address token_, AggregatorV3Interface oracle_) internal {
+        require(token_ != address(0), "token cannot be zero address");
+        require(oracle_ != AggregatorV3Interface(address(0)), "oracle cannot be zero address");
+        if(Oracles[token_] == oracle_ && Ratios[token_] != 0) {
+            revert("token has already been added as collateral");
+        }
+        if(Oracles[token_] == oracle_ && Ratios[token_] == 0)  {
+            revert("token added but ratio not set");
+        } else{
+            Collaterals.push(token_);
+            Oracles[token_] = oracle_;
+        }
+    }
+
+    /// @dev A token must be added to the list of accepted collateral before a ratio can be set for it
+    function _setRatio(address token_, uint256 rad) internal {
+        require(token_ != address(0), "token cannot be zero address");
+        require(rad > 0, "ratio must be greater than 0");
+        require(rad < 1e27, "ratio must be less than 1");
+        require(Oracles[token_] != AggregatorV3Interface(address(0)), "token has not been added as collateral");
+
+        Ratios[token_] = rad;
+    }
+
+    
+    /// @notice allows authorized user to liquidate under collateralized Users
+    function _liquidate(address user) internal {
+        require(_ltvRatio(user) < _maxRatio(user), "specified user can't be liquidated");
```

Comment:
Here you are calculating `ratioedDeposits` twice, once for `ltvRatio` and once for `maxRatio`. You can merge `_ltvRatio(user) < _maxRatio(user)` into a single `isCollateralized` function and save a lot of gas.